numpy/_core/defchararray.py

```diff
@@ -2093,24 +2090,20 @@ def __new__(subtype, shape, itemsize=1, unicode=False, buffer=None,
                                    order=order)
         if filler is not None:
             self[...] = filler
-        _globalvar = 0
-        return self
 
-    def __array_prepare__(self, arr, context=None):
-        # When calling a ufunc, we return a chararray if the ufunc output
-        # is a string-like array, or an ndarray otherwise
-        if arr.dtype.char in "SUbc":
-            return arr.view(type(self))
-        return arr
+        return self
 
     def __array_wrap__(self, arr, context=None):
+        # When calling a ufunc (and some other functions), we return a
+        # chararray if the ufunc output is a string-like array,
+        # or an ndarray otherwise
         if arr.dtype.char in "SUbc":
             return arr.view(type(self))
         return arr
 
     def __array_finalize__(self, obj):
         # The b is a special case because it is used for reconstructing.
-        if not _globalvar and self.dtype.char not in 'SUbc':
+        if self.dtype.char not in 'SUbc':
             raise ValueError("Can only create a chararray from string data.")
```

Comment:
`mmap` also had such a global once upon a time.  I don't get it.  If there is a super strange reason for why it's needed let's put it back when it shows up...

Reply:
Strange, indeed. It looks like it protects the actual creation of the array (which will call `__array_finalize__`), but I'm not sure what that is needed. And tests pass...